lib/ansible/modules/cloud/docker/docker_plugin.py

```diff
@@ -0,0 +1,297 @@
+    module has been superseded by L(docker,https://pypi.org/project/docker/)
+    (see L(here,https://github.com/docker/docker-py/issues/1310) for details).
+    Version 2.1.0 or newer is only available with the C(docker) module."
+  - "Docker API >= 1.25"
+'''
+
+EXAMPLES = '''
+- name: Install a plugin
+  docker_plugin:
+    name: plugin_one
+
+- name: Remove a plugin
+  docker_plugin:
+    name: plugin_one
+    state: absent
+
+- name: Install a plugin with options
+  docker_plugin:
+    name: weaveworks/net-plugin:latest_release
+    alias: weave-net-plugin
+    plugin_options:
+      IPALLOC_RANGE: "10.32.0.0/12"
+      WEAVE_PASSWORD: XXXXXXXX
+'''
+
+RETURN = '''
+facts:
+    description: Plugin inspection results for the affected plugin.
+    returned: success
+    type: dict
+    sample: {}
+'''
+
+try:
+    from docker.errors import APIError, NotFound
+    from docker.models.plugins import Plugin
+    from docker import DockerClient
+except ImportError:
+    # missing docker-py handled in ansible.module_utils.docker_common
+    pass
+
+from ansible.module_utils.docker.common import DockerBaseClass, AnsibleDockerClient
+from ansible.module_utils.six import iteritems, text_type
+
+
+class TaskParameters(DockerBaseClass):
+    def __init__(self, client):
+        super(TaskParameters, self).__init__()
+        self.client = client
+
+        self.name = None
+        self.alias = None
+        self.plugin_options = None
+        self.debug = None
+
+        for key, value in iteritems(client.module.params):
+            setattr(self, key, value)
+
+
+def prepare_options(options):
+    return ['%s=%s' % (k, v if v is not None else "") for k, v in iteritems(options)] if options else []
+
+
+def parse_options(options_list):
+    return dict((k, v) for k, v in map(lambda x: x.split('=', 1), options_list)) if options_list else {}
+
+
+class DockerPluginManager(object):
+
+    def __init__(self, client):
+        self.client = client
+
+        self.dclient = DockerClient()
+        self.dclient.api = client
+
+        self.parameters = TaskParameters(client)
+        self.check_mode = self.client.check_mode
+        self.results = {
+            u'changed': False,
+            u'actions': []
+        }
+        self.diff = self.client.module._diff
+
+        self.existing_plugin = self.get_existing_plugin()
+
+        state = self.parameters.state
+        if state == 'present':
+            self.present()
+        elif state == 'absent':
+            self.absent()
+        elif state == 'enable':
+            self.enable()
+        elif state == 'disable':
+            self.disable()
+
+    def get_plugin_name(self):
+        return self.parameters.alias or self.parameters.name
+
+    def get_existing_plugin(self):
+        name = self.get_plugin_name()
+        try:
+            plugin = self.dclient.plugins.get(name)
+        except NotFound:
+            return None
+        except APIError as e:
+            self.client.fail(text_type(e))
```

Comment:
It is usually a good idea to prepend some text so that it is easier to find out where an error was raised (in case it's not clear from the error message).